src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -105,6 +105,26 @@
      */
     private final Cache<String, ActiveDirectoryGroupDetails> groupCache;
 
+    /**
+     * Sets com.sun.jndi.ldap.connect.timeout
+     */
+    private final String ldapConnectTimeout;
+
+    /**
+     * Sets com.sun.jndi.ldap.read.timeout
+     */
+    private final String ldapReadTimeout;
+
+    /**
+     * Timeout if no connection after 30 seconds
+     */
+    private static final long DEFAULT_LDAP_CONNECTION_TIMEOUT = 30000;
```

Comment:
if someone has set the system property for their env then you are trashing what they have carefully set already...